sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorStateStore.java

```diff
@@ -193,14 +192,45 @@ public void storeTasks(Collection<Protos.TaskInfo> tasks) throws StateStoreExcep
 
     @Override
     public void storeStatus(Protos.TaskStatus status) throws StateStoreException {
-        String taskName;
+        Optional<String> optionalTaskName = Optional.empty();
+
+        //TODO(MB): keep the commented code below: original approach for getting TaskName
+        /* We used to get TaskName from TaskID's prefix, instead of comparing TaskIDs.
+           TaskID is our unique identifier, can not change, but we can modify TaskName
+
+           Getting TaskName from prefix may be more efficient, however it prevents us from changing TaskName.
```

Comment:
If you think toTasName() has no performance benefit and no further effort needed in this, I will delete the comments